### ISSUE TITLE
Cannot write to unmasked output

### ISSUE THREAD

**Issue by JulioHC00** [lightkurve/lightkurve#1168]
#### Problem description

When trying to flatten a lightcurve I have just downloaded, I get 

`TypeError: cannot write to unmasked output`

I have tried passing the argument `mask=np.zeros([len(lc.time)], dtype=bool)` to see if it would do anything but the same happens.
#### Example

```python
import lightkurve as lk
import matplotlib.pyplot as plt
import numpy as np

search_results = lk.search_lightcurve("HD 39091", mission="TESS", exptime="short", sector=39)

lc = search_results.download()

lc_f = lc.flatten(window_length=901))
```

#### Expected behavior

Get a flattened lightcurve.

#### Environment

-  platform (e.g. Linux, OSX, Windows): Linux, Ubuntu 21.04
-  lightkurve version (e.g. 1.0b6): 2.0.11
-  installation method (e.g. pip, conda, source): pip

EDIT: Even more weird is that this issue happens only on one computer but not on a different one. I've tried uninstalling lightkurve and reinstalling and the same happens.

**Comment by Scratchcat1** [comment]
I encountered a similar error with flatten which is caused by this bug in AstroPy: https://github.com/lightkurve/lightkurve/issues/1161

I managed to fix it by installing `astropy==4.3.1`.

For reference my error output was:
```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Input In [7], in <module>
      2     corrected_lc = lc.normalize().flatten(window_length=101,mask=np.zeros([len(lc.time)], dtype=bool))
      3     return corrected_lc
----> 5 stitched_lc = lcfs.PDCSAP_FLUX.stitch(corrector_func=my_custom_corrector_func)
      6 stitched_lc = stitched_lc.remove_nans()
      7 stitched_lc.scatter()

File ~/Documents/test/test-lightkurve/venv/lib/python3.9/site-packages/lightkurve/collections.py:195, in LightCurveCollection.stitch(self, corrector_func)
    193 with warnings.catch_warnings():  # ignore "already normalized" message
    194     warnings.filterwarnings("ignore", message=".*already.*")
--> 195     lcs = [corrector_func(lc) for lc in self]
    197 # Address issue #954: ignore incompatible columns with the same name
    198 columns_to_remove = set()

File ~/Documents/test/test-lightkurve/venv/lib/python3.9/site-packages/lightkurve/collections.py:195, in <listcomp>(.0)
    193 with warnings.catch_warnings():  # ignore "already normalized" message
    194     warnings.filterwarnings("ignore", message=".*already.*")
--> 195     lcs = [corrector_func(lc) for lc in self]
    197 # Address issue #954: ignore incompatible columns with the same name
    198 columns_to_remove = set()

Input In [7], in my_custom_corrector_func(lc)
      1 def my_custom_corrector_func(lc):
----> 2     corrected_lc = lc.normalize().flatten(window_length=101,mask=np.zeros([len(lc.time)], dtype=bool))
      3     return corrected_lc

File ~/Documents/test/test-lightkurve/venv/lib/python3.9/site-packages/lightkurve/lightcurve.py:732, in LightCurve.flatten(self, window_length, polyorder, return_trend, break_tolerance, niters, sigma, mask, **kwargs)
    730     mask = deepcopy(~mask)
    731 # No NaNs
--> 732 mask &= np.isfinite(self.flux)
    733 # No outliers
    734 mask &= np.nan_to_num(np.abs(self.flux - np.nanmedian(self.flux))) <= (
    735     np.nanstd(self.flux) * sigma
    736 )

File ~/Documents/test/test-lightkurve/venv/lib/python3.9/site-packages/astropy/utils/masked/core.py:674, in MaskedNDArray.__array_ufunc__(self, ufunc, method, *inputs, **kwargs)
    671 if m is None:
    672     # TODO: allow writing to unmasked output if nothing is masked?
    673     if d is not None:
--> 674         raise TypeError('cannot write to unmasked output')
    675 elif out_mask is None:
    676     out_mask = m

TypeError: cannot write to unmasked output

```

**Comment by astrobel** [comment]
I get the same error when using the CBV corrector. As below:

```
import lightkurve as lk

target = lk.search_lightcurve('TIC 159449703').download().select_flux('sap_flux').remove_nans()

cbvc = lk.correctors.CBVCorrector(target)
cbv_type = ['SingleScale', 'Spike']
cbv_indices = [np.arange(1,9), 'ALL']
cbvc.correct(cbv_type=cbv_type, cbv_indices=cbv_indices)
```

```
TypeError                                 Traceback (most recent call last)
Input In [27], in <cell line: 1>()
----> 1 cbvc.correct(cbv_type=cbv_type, cbv_indices=cbv_indices)
      2 cbvc.diagnose()

File ~/.local/lib/python3.10/site-packages/lightkurve/correctors/cbvcorrector.py:458, in CBVCorrector.correct(self, cbv_type, cbv_indices, ext_dm, cadence_mask, alpha_bounds, target_over_score, target_under_score, max_iter)
    448 self.optimization_params = {'alpha_bounds': alpha_bounds,
    449                             'target_over_score': target_over_score,
    450                             'target_under_score': target_under_score,
    451                             'max_iter': max_iter,
    452                             'cadence_mask': cadence_mask,
    453                             'over_metric_nSamples': 1}
    455 #***
    456 # Use scipy.optimize.minimize_scalar
    457 # Minimize the introduced metric
--> 458 minimize_result = minimize_scalar(self._goodness_metric_obj_fun, method='Bounded',
    459         bounds=alpha_bounds,
    460         options={'maxiter':max_iter, 'disp': False})
    462 # Re-fit with final alpha value
    463 # (scipy.optimize.minimize_scalar does not exit with the final fit!)
    464 self._goodness_metric_obj_fun(minimize_result.x)

File ~/.local/lib/python3.10/site-packages/scipy/optimize/_minimize.py:804, in minimize_scalar(fun, bracket, bounds, args, method, tol, options)
    802     if isinstance(disp, bool):
    803         options['disp'] = 2 * int(disp)
--> 804     return _minimize_scalar_bounded(fun, bounds, args, **options)
    805 elif meth == 'golden':
    806     return _minimize_scalar_golden(fun, bracket, args, **options)

File ~/.local/lib/python3.10/site-packages/scipy/optimize/optimize.py:2024, in _minimize_scalar_bounded(func, bounds, args, xatol, maxiter, disp, **unknown_options)
   2022 rat = e = 0.0
   2023 x = xf
-> 2024 fx = func(x, *args)
   2025 num = 1
   2026 fmin_data = (1, xf, fx)

File ~/.local/lib/python3.10/site-packages/lightkurve/correctors/cbvcorrector.py:799, in CBVCorrector._goodness_metric_obj_fun(self, alpha)
    797 self._set_prior_width(sigma)                
    798 # Use RegressionCorrector.correct for the actual fitting
--> 799 self.correct_regressioncorrector(self.design_matrix_collection, 
    800     cadence_mask=self.optimization_params['cadence_mask'])
    802 # Do not compute and ignore if target score < 0
    803 if (self.optimization_params['target_over_score'] > 0):

File ~/.local/lib/python3.10/site-packages/lightkurve/correctors/cbvcorrector.py:493, in CBVCorrector.correct_regressioncorrector(self, design_matrix_collection, **kwargs)
    487 """ Pass-through method to gain access to the superclass 
    488 RegressionCorrector.correct() method.
    489 """
    491 # All this does is call the superclass 'correct' method as pass the
    492 # input arguments.
--> 493 return super(CBVCorrector, self).correct(design_matrix_collection, **kwargs)

File ~/.local/lib/python3.10/site-packages/lightkurve/correctors/regressioncorrector.py:257, in RegressionCorrector.correct(self, design_matrix_collection, cadence_mask, sigma, niters, propagate_errors)
    255     model = u.Quantity(model, unit=self.lc.flux.unit)
    256     residuals = self.lc.flux - model
--> 257     self.outlier_mask |= sigma_clip(residuals, sigma=sigma).mask
    258     log.debug(
    259         "correct(): iteration {}: clipped {} cadences"
    260         "".format(count, self.outlier_mask.sum())
    261     )
    263 self.coefficients = coefficients

File ~/.local/lib/python3.10/site-packages/astropy/utils/masked/core.py:674, in MaskedNDArray.__array_ufunc__(self, ufunc, method, *inputs, **kwargs)
    671 if m is None:
    672     # TODO: allow writing to unmasked output if nothing is masked?
    673     if d is not None:
--> 674         raise TypeError('cannot write to unmasked output')
    675 elif out_mask is None:
    676     out_mask = m

TypeError: cannot write to unmasked output
```

I was not able to install an older version of Astropy as a workaround, due to another incompatibility.

**Comment by barentsen** [comment]
I tried running the example posted by @JulioHC00 at the start of this issue using the latest version of Lightkurve (v2.1.1) and found it to work correctly:

<img width="893" alt="Screen Shot 2022-03-28 at 8 15 23 PM" src="https://user-images.githubusercontent.com/817669/160526097-266242f4-5ad6-4da0-8c95-29895f3912f0.png">

**Comment by barentsen** [comment]
Likewise, I am also able to run the CBVCorrector example posted by @astrobel above with Lightkurve v2.1.1.  (This is of course consistent with @astrobel's own findings.)

I am going to close this issue because it looks like the issues reported here have been resolved in Lightkurve v2.1.1.  I do encourage anyone who encounters these errors with v2.1.1 or later to comment below or open a new issue.

As a reminder, you can upgrade Lightkurve using:
```
python -m pip install --upgrade lightkurve
```

And you can check the version number as follows:
```python
import lightkurve
print(lightkurve.__version__)
```
